testcases.py

```diff
@@ -1262,7 +1265,16 @@ def check(self) -> TestResult:
             self._server_trace()._get_direction_filter(Direction.FROM_SERVER) + " quic"
         )
 
-        ips = list(set(getattr(p["ip"], "dst") for p in tr_server))
+        ips = list(
+            set(
+                (
+                    getattr(p["ipv6"], "dst")
+                    if "IPV6" in str(p.layers)
+                    else getattr(p["ip"], "dst")
+                )
+                for p in tr_server
+            )
+        )
```

Comment:
This is getting complex enough that I'd recommend doing this in multiple lines, for readability.
```
ips = set()
for p in tr_server:
    ip_tag = "ip"
    if "IPV6" in str(p.layers): ip_tag = "ipv6"
    ips.add(getattr(ip_tag, "dst"))
```
You also don't need to convert this to a list, since the operations below ought to work with set as well.